### PR TITLE
added parsing and checking functions docblocks

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**
This should fix this issue #301 
Is it a BC?

### CODE REVIEW

File: src/Analyzer/FileVisitor.php [hunk 1]
```diff
@@ -133,6 +127,12 @@ public function enterNode(Node $node): void
             );
         }
 
+        if (null !== $this->classDescriptionBuilder && null !== $node->getDocComment()) {
```
Comment (micheleorselli) [inline src/Analyzer/FileVisitor.php:130]: do we need the null check on `$this->classDescriptionBuilder`? I find it a bit confusing, as we are already using `$this->classDescriptionBuilder` a few lines above without any check

**Review by micheleorselli — COMMENTED** [review]
Noice! Left a minor, non blocking comment

File: src/Analyzer/FileVisitor.php (hunk 1 above)
Comment (AlessandroMinoccheri) [reply src/Analyzer/FileVisitor.php:130]: I have got different errors without this check

Comment (micheleorselli) [reply src/Analyzer/FileVisitor.php:130]: I am not sure I got what you mean but just out of curiosity: doesn't the check on line 28 prevent from having a null in `$this->classDescriptionBuilder` here?

And just to be super clear: all good from me, feel free to merge it

**Review by fain182 — APPROVED** [review]
:+1:

File: src/Analyzer/FileVisitor.php (hunk 1 above)
Comment (AlessandroMinoccheri) [reply src/Analyzer/FileVisitor.php:130]: The check on line 28 is inside an if that finish before the new check, this check is more generic

Comment (AlessandroMinoccheri) [reply src/Analyzer/FileVisitor.php:130]: split the if @fain182 ? What is confusing in this part?

**Pull request merged**